cmd/podman-mac-helper/main.go

```diff
@@ -0,0 +1,331 @@
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	dockerSock = "/var/run/docker.sock"
+	fail       = "NO"
+	success    = "OK"
+)
+
+const launchConfig = `<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>Label</key>
+	<string>com.github.containers.podman.helper-{{.User}}</string>
+	<key>ProgramArguments</key>
+	<array>
+		<string>{{.Program}}</string>
+		<string>service</string>
+		<string>{{.Target}}</string>
+	</array>
+	<key>inetdCompatibility</key>
+	<dict>
+		<key>Wait</key>
+		<false/>
+	</dict>
+	<key>UserName</key>
+	<string>root</string>
+	<key>Sockets</key>
+	<dict>
+		<key>Listeners</key>
+		<dict>
+			<key>SockFamily</key>
+			<string>Unix</string>
+			<key>SockPathName</key>
+			<string>/private/var/run/podman-helper-{{.User}}.socket</string>
+			<key>SockPathOwner</key>
+			<integer>{{.UID}}</integer>
+			<key>SockPathMode</key>
+			<!-- SockPathMode takes base 10 (384 = 0600) -->
+			<integer>384</integer>
+			<key>SockType</key>
+			<string>stream</string>
+		</dict>
+	</dict>
+</dict>
+</plist>
+`
+
+type launchParams struct {
+	Program string
+	User    string
+	UID     string
+	Target  string
+}
+
+// Note, this code is security sensitive since it runs under privilege.
+// Limit actions to what is strictly necessary, and take use appropriate
+// safeguards
+//
+// After installation the service call is ran under launchd in a nowait
+// inetd style fashion, so stdin, stdout, and stderr are all pointing to
+// an accepted connection
+//
+// This service is installed once per user and will redirect
+// /var/run/docker to the fixed user-assigned unix socket location.
+//
+// Control communication is restricted to each user specific service via
+// unix file permissions
+
+func main() {
+	usageFmt := "Usage: %s [install | uninstall | service]\n"
+	prog, err := getProgram()
+	if err != nil {
+		prog = "unknown"
+	}
+
+	if len(os.Args) < 2 {
+		fmt.Printf(usageFmt, filepath.Base(prog))
+		os.Exit(1)
+	}
+
+	if os.Geteuid() != 0 {
+		fmt.Printf("Must be ran as root via sudo or osascript. Run the following:\nsudo %s %s\n", prog, os.Args[1])
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case "install":
+		err = install()
+	case "uninstall":
+		err = uninstall()
+	case "service":
+		os.Exit(service())
+	default:
+		fmt.Println(usageFmt, filepath.Base(prog))
+	}
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %s\n", err.Error())
+		os.Exit(2)
+	}
+
+	os.Exit(0)
+}
+
+func getProgram() (string, error) {
+	exec, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	exec, err = filepath.EvalSymlinks(exec)
+	return exec, err
+}
+
+func getUserInfo(name string) (string, string, string, error) {
+	// We exec id instead of using user.Lookup to remain compat
```

Comment:
@n1hility  I could be wrong but wouldn't `-tags osusergo` help with this?

Reply:
hmm probably not.

Reply:
@matejvasek yeah unfortunately all non CGO builds use a generic Unix impl that parses passwd files, which on Mac are limited to just base system accounts. I hope one day golang gets a generic limited syscall wrapper like they have for windows. For this usage though the overhead is not a factor, so I figured no need for a cross compile build chain setup.